src/app/components/shell/microsurvey-popup/adoption-content.js

```diff
@@ -1,21 +1,65 @@
 import React, {useRef} from 'react';
 import useUserContext from '~/contexts/user';
-import useAdoptions from '~/models/renewals';
+import {useLocation} from 'react-router-dom';
+import {useToggle} from '~/components/jsx-helpers/jsx-helpers.jsx';
+import cookie from '~/helpers/cookie';
+
+const DISMISSED_KEY = 'renewal_dialog_dismissed';
+const YESTERDAY = Date.now() - 60 * 60 * 24 * 1000;
+
+function useCookieKey(key) {
+    return React.useReducer(
+        (_, value) => {
+            cookie.setKey(key, value);
+            return value ? value : '0';
+        },
+        cookie.hash[key]
+    );
+}
```

Comment:
I typically prefer using local state to store values, but maybe this also needs to be read from the server?

I've used [useLocalStorage](https://github.com/nas5w/use-local-storage) before but would be pretty easy to write your own version